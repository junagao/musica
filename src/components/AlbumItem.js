import React from 'react';
import PropTypes from 'prop-types';

import './AlbumItem.scss';

const AlbumItem = ({ artist, artwork, album }) => (
  <li className="album-item">
    <img src={artwork.replace('100x100', '200x200')} alt="" />
    <div className="album-details">
      <p className="album-title">{album}</p>
      <p className="album-artist">{artist}</p>
    </div>
  </li>
);

AlbumItem.propTypes = {
  artist: PropTypes.string.isRequired,
  artwork: PropTypes.string.isRequired,
  album: PropTypes.string.isRequired,
};

export default AlbumItem;
