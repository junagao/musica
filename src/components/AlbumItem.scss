@import '../variables';

.album-item {
  border-radius: 0.25rem;
  border: 1px solid transparent;
  width: 150px;
  animation: fadeIn 1s;
}

img {
  border-radius: 0.25rem;
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),
    0 4px 6px -2px rgba(0, 0, 0, 0.05);
  width: 100%;
}

.album-details {
  padding: 0.25rem 0rem;
}

.album-title {
  color: $color-album-title;
  font-size: 0.95rem;
  font-weight: 600;
}

.album-artist {
  color: $color-album-artist;
  font-size: 0.85rem;
  font-weight: 400;
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
