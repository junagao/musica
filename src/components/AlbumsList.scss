@import '../variables';

.albums-list-container {
  padding: 2rem 0rem 0rem 0rem;
}

.albums-list-header {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

h2 {
  color: $color-h2;
  font-size: 1.5rem;
}

.view-type-container {
  justify-self: end;
}

.view-btn {
  color: $color-lighter;
  margin-left: 0.75rem;
  font-size: 1.5rem;
  cursor: pointer;
  outline: none;
  border: 0;
  text-decoration: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  transition: all ease-out 0.3s;
  &:hover,
  &:focus {
    color: $color-darker;
    transition: all ease-in 0.3s;
  }
}

.albums-list {
  display: grid;
  grid-column-gap: 2rem;
  grid-row-gap: 1.5rem;
  padding: 1rem 0rem 2rem 0rem;
  justify-content: space-between;
  @media (min-width: $tabletP) {
    grid-column-gap: 1rem;
  }
  @media (min-width: $laptop) {
    grid-column-gap: 4rem;
  }
  @media (min-width: $desktop) {
    grid-column-gap: 6rem;
  }
  @media (min-width: $desktopHD) {
    grid-column-gap: 6rem;
  }
}

.grid {
  grid-template-columns: repeat(auto-fill, 150px);
  animation: fadeIn 1s;
  @media (min-width: $desktopHD) {
    grid-template-columns: repeat(auto-fill, 175px);
  }
}

.list {
  grid-template-rows: 1fr 1fr;
  grid-auto-rows: row;
  animation: fadeIn 1s;
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
