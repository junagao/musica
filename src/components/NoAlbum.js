import React from 'react';

import './NoAlbum.scss';

const NoAlbum = () => <div className="no-album">Sorry, no album found.</div>;

export default NoAlbum;
