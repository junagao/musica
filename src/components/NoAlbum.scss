.no-album {
  font-size: 1.25rem;
  padding: 2rem 0rem;
}
