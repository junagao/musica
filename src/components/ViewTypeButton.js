import React from 'react';
import PropTypes from 'prop-types';

import './ViewTypeButton.scss';

const ViewTypeButton = ({ onChangeView, viewType, icon }) => (
  <button
    type="button"
    className="view-btn"
    onClick={() => onChangeView(viewType)}
  >
    {icon}
  </button>
);

ViewTypeButton.propTypes = {
  onChangeView: PropTypes.func.isRequired,
  viewType: PropTypes.string.isRequired,
  icon: PropTypes.element.isRequired,
};

export default ViewTypeButton;
