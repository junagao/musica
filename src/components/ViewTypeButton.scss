@import '../variables';

.view-btn {
  color: $color-lighter;
  background-color: transparent;
  margin-left: 0.75rem;
  font-size: 1.5rem;
  cursor: pointer;
  outline: none;
  border: 0;
  text-decoration: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  transition: all ease-out 0.3s;
  &:hover,
  &:focus {
    color: $color-darker;
    transition: all ease-in 0.3s;
  }
}
