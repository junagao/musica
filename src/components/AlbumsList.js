import React from 'react';
import PropTypes from 'prop-types';

import AlbumItem from './AlbumItem';

const AlbumsList = ({ albums }) => {
  return (
    <div>
      <ul className="album-list">
        {albums.map((album) => {
          const {
            collectionId,
            artistName,
            artworkUrl100,
            collectionName,
          } = album;
          return (
            <AlbumItem
              key={collectionId}
              artwork={artworkUrl100}
              artist={artistName}
              album={collectionName}
            />
          );
        })}
      </ul>
    </div>
  );
};

AlbumsList.propTypes = {
  albums: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.number,
      artwork: PropTypes.string,
      artist: PropTypes.string,
      album: PropTypes.string,
    }),
  ).isRequired,
};

export default AlbumsList;
