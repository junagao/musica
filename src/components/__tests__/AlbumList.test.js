import React from 'react';
import { shallow } from 'enzyme';

import AlbumsList from 'components/AlbumsList';
import ViewTypeButton from 'components/ViewTypeButton';
import AlbumItem from 'components/AlbumItem';
import NoAlbum from 'components/NoAlbum';

describe('AlbumList', () => {
  let wrapper;
  const albums = [
    {
      collectionId: 793928184,
      artistName: 'Tycho',
      collectionName: 'Awake',
      artworkUrl100:
        'https://is4-ssl.mzstatic.com/image/thumb/Music/v4/b9/76/bc/b976bc5d-2c25-d79d-4057-67bc47b4ab6d/source/100x100bb.jpg',
    },
    {
      collectionId: 1461523692,
      artistName: 'Tycho',
      collectionName: 'Weather',
      artworkUrl100:
        'https://is1-ssl.mzstatic.com/image/thumb/Music123/v4/a9/bf/0c/a9bf0c63-fb1d-b3ee-315f-40d829f0159c/source/100x100bb.jpg',
    },
  ];

  beforeEach(() => {
    wrapper = shallow(
      <AlbumsList albums={albums} onChangeView={jest.fn()} noResults={false} />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 3 div elements', () => {
    expect(wrapper.find('div').length).toEqual(3);
  });

  it('should render the first div element with className of albums-list-container', () => {
    expect(
      wrapper
        .find('div')
        .at(0)
        .hasClass('albums-list-container'),
    ).toBeTruthy();
  });

  it('should render the second div element with className of albums-list-header', () => {
    expect(
      wrapper
        .find('div')
        .at(1)
        .hasClass('albums-list-header'),
    ).toBeTruthy();
  });

  it('should render one h2 element with the value `Albums`', () => {
    expect(wrapper.find('h2').length).toEqual(1);
    expect(wrapper.find('h2').text()).toEqual('Albums');
    expect(wrapper.containsMatchingElement(<h2>Albums</h2>)).toBeTruthy();
  });

  it('should render the third div element with className of view-type-container', () => {
    expect(
      wrapper
        .find('div')
        .at(2)
        .hasClass('view-type-container'),
    ).toBeTruthy();
  });

  it('should render two view type buttons', () => {
    expect(wrapper.find(ViewTypeButton).length).toEqual(2);
  });

  it('should render a unordered list with className albums-list and grid', () => {
    expect(wrapper.find('ul').length).toEqual(1);
    expect(wrapper.find('ul').hasClass('albums-list')).toBeTruthy();
    expect(wrapper.find('ul').hasClass('grid')).toBeTruthy();
  });

  it('should render two instances of <AlbumItem />', () => {
    expect(wrapper.find(AlbumItem).length).toEqual(2);
  });

  it('should render NoAlbum if no search results is found', () => {
    wrapper.setProps({ noResults: true });
    expect(wrapper.find(NoAlbum).length).toEqual(1);
  });
});
