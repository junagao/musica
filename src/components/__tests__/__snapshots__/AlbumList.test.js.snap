// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumList should render correctly given the required props 1`] = `
<div
  className="albums-list-container"
>
  <div
    className="albums-list-header"
  >
    <h2>
      Albums
    </h2>
    <div
      className="view-type-container"
    >
      <ViewTypeButton
        icon={<FaThList />}
        onChangeView={[MockFunction]}
        viewType="list"
      />
      <ViewTypeButton
        icon={<FaThLarge />}
        onChangeView={[MockFunction]}
        viewType="grid"
      />
    </div>
  </div>
  <ul
    className="albums-list grid"
  >
    <AlbumItem
      album="Awake"
      artist="Tycho"
      artwork="https://is4-ssl.mzstatic.com/image/thumb/Music/v4/b9/76/bc/b976bc5d-2c25-d79d-4057-67bc47b4ab6d/source/100x100bb.jpg"
      key="793928184"
    />
    <AlbumItem
      album="Weather"
      artist="Tycho"
      artwork="https://is1-ssl.mzstatic.com/image/thumb/Music123/v4/a9/bf/0c/a9bf0c63-fb1d-b3ee-315f-40d829f0159c/source/100x100bb.jpg"
      key="1461523692"
    />
  </ul>
</div>
`;
